### ISSUE TITLE
Fix Code style for non-public, non-static fields

### ISSUE THREAD

**Issue by david-allison-1** [ankidroid/Anki-Android#8387]
An easy, but slightly repetitive first issue.

I've added a pull request: #8370 which ensures that all fields match the following lint check: https://github.com/ankidroid/Anki-Android/wiki/Code-style#non-public-non-static-field-names-should-start-with-m

The problem is that our code has ~370 instances where we have variables which don't match this rule. 

We need to fix these before we turn on this lint rule

**This does not all need to be done in a single PR**

----

To fix this:

* checkout my pull request branch: 
  * Suggested method: 
  * Define a git remote (I use `upstream`) which points to this repository
     *  `git remote add upstream https://github.com/ankidroid/Anki-Android.git`
  * ```bash
    git fetch upstream pull/8370/head:8370-fix
    git checkout 8370-fix
    ```
  * OR: (after PRs have been added: `git cherry-pick` the commit adding the lint check onto the current `master` branch
* Find the issues:
  * ⚠️ Ignore (`@SuppressLint("FieldNamingPatternDetector")`) for the classes with issues under `wildplot.android`
  * best option: https://github.com/ankidroid/Anki-Android/issues/8387#issuecomment-809722269
  * **OR:**
  * `gradlew lintRelease`
  * open the html file linked in the lintRelease output 
* Fix each issue (rename the variable to start with `m`)
* Commit regularly - likely once per file
* Generate a branch without the lint check 
   * ` git cherry-pick` your commits to a new branch
   * OR: rebase and drop the commit containing the lint check
* Submit a pull request when you feel you've worked enough

We cherry pick (or rebase), as we don't want the commit to add the lint check in history before all the issues are fixed

**Comment by david-allison-1** [comment]
@Arthur-Milchior, just want to confirm that this seems like a good way to go about this

* Are you happy with the granularity of commits?
* Would you accept pull requests fixing some, but not all of the issues, or would you want one big PR?

**Comment by Arthur-Milchior** [comment]
I'm very fine with PR doing only some issues. If someone starts the work, do one hour of work and stop there, I'm totally fine accepting it.

Note that if you click on a variable you can use shift+F6 to rename the variable and all instances of it. That may really hepl.

If the build fail with this lint rule, then I would just try to build, and use F7 to move from one failure to the next one. This way, I can quickly correct all failures without needing to do anytihng more complex than clicking on build

**Comment by david-allison-1** [comment]
They're shown as fatal errors after a `gradlew lintRelease`, but don't break the build.

`AstractFlashCardViewer.lastCrashingCardId` is the first good example

should be `AstractFlashCardViewer.mLastCrashingCardId`

Run Inspection By Name: `Ctrl+Alt+Shift+I` -> "field naming": 
![image](https://user-images.githubusercontent.com/62114487/112901542-ceb36400-90dc-11eb-8637-d7f35739be84.png)

Wait

![image](https://user-images.githubusercontent.com/62114487/112901662-f1de1380-90dc-11eb-9b48-e451e3c9fa39.png)

**Comment by Arnold2381** [comment]
![image](https://user-images.githubusercontent.com/46752548/112933082-c4c14d80-913c-11eb-9899-6cdbd922fd86.png)
@David I am not getting the non public non static inspection name

**Comment by david-allison-1** [comment]
@Arnold2381, do you plan to work on this?

**Comment by Arnold2381** [comment]
@david-allison-1 sorry for the late reply! ya sure I will start working on this, if I will face issues will ask you.

**Comment by Arnold2381** [comment]
@david-allison-1  I ran gradlew lintRelease getting version error is there a short way to solve it ?



ERROR: AnkiDroid builds with JVM LTS and current releases (currently major version 8, 11, or 14).
Incompatible major version detected: '15'

**Comment by david-allison-1** [comment]
You should be running a LTS version of Java. You'll need to download it or switch to it if it's already installed

**Comment by Arnold2381** [comment]
@david-allison-1 umm okay thanks!!

**Comment by mikehardy** [comment]
Actually, the error message says LTS or current release, and I think I put that in there? Although I think you were involved. It appears Java15 has come out in the meantime. But wait, Java16 is out now as of a week or so ago

So there is a new issue to log: "JDK Current is now 16, verify AnkiDroid may be developed successfully using it and alter the Gradle JDK check to allow LTS versions and 16"

@Arnold2381 for now, definitely use "an LTS version", which is either 8 or 11.

**Comment by Arnold2381** [comment]
@mikehardy Yea exactly Java 16 is out too!!
Yea I downloaded Java 11 and switched to that after which I was successfully able to run the Lint checks and generate a report of the errors for the same.
Should I log the issue?

**Comment by mikehardy** [comment]
I opened #8434 to track JDK16 issue, it's completely separate from this and should be pursued independently

**Comment by mikehardy** [comment]
Not closed yet - just one file done so far

**Comment by Arnold2381** [comment]
Yea working on others too! @mikehardy

**Comment by david-allison-1** [comment]
@Arnold2381 Just to note: I've removed a lot of these warnings for `wildplot` in #8437

In future, to ensure that work hasn't been duplicated, you'll want to:

* Check out master & Pull latest changes
* Make a branch
* `git cherry-pick bb93b0af726ab4c37008fabc415b961c4568d6dc`
* You can then run inspections, and drop the commit later

**Comment by Arnold2381** [comment]
@david-allison-1 noted. But I think for cherry pick the working tree needs to be clean right?

I am already done making a lot of changes around 10-15 files was going to commit that. 

Next time I will make sure.

**Comment by david-allison-1** [comment]
I'd find cherry picking to be the easiest way to keep up, but if you've got a workflow that works, don't disrupt it, do what works.

Thanks so much!

**Comment by Arnold2381** [comment]
Sorry! Will surely use cherry pick next time! thanks for the suggestion
And also should I make a PR now? Or solve all the issues and make a PR at the end?

**Comment by david-allison-1** [comment]
Don't worry at all, do what works for you. Definitely don't apologize!

If I were you, I'd break it into a pull request per day.

Also has the added benefit of more GitHub contributions 😉

**Comment by Arnold2381** [comment]
Haha!! Okkk

**Comment by mikehardy** [comment]
you can `git stash` then cherry-pick then `git pop` - I do that a lot, but more frequently I stash, then pull/rebase against main branch then stash pop to stay up to date

alternatively you can commit your changes with a WIP (work in progress) commit message (so it's obvious), pull/rebase, then `git reset --soft HEAD~1 && git reset` to pop your WIP commit off and unstage the changes

git fun ;-)

**Comment by Arnold2381** [comment]
![image](https://user-images.githubusercontent.com/46752548/113446840-54554d80-9416-11eb-8821-a4ad538b1a9c.png)

Thanks for the explanation, that was a great one. git checkout meme 😄😄😄

**Comment by david-allison-1** [comment]
To discuss: `    boolean _v2sched;`

https://github.com/ankidroid/Anki-Android/pull/8451#discussion_r606666268

**Comment by david-allison-1** [comment]
Closed in #8532

@Arnold2381 Thank you so much for all the effort!!!! I know it was a slog, but the rewards are massive: it means that we can spend less time reviewing, developers get less "nitpicks" on their code reviews, and they get instant feedback. It makes the process of development much more pleasant for everyone involved. 

![](https://media.giphy.com/media/ZfK4cXKJTTay1Ava29/giphy.gif)

**Comment by Arnold2381** [comment]
My pleasure @david-allison-1 !!
Thanks to you too :)
Hehe, nice one.
